plugin/documents.py

```diff
@@ -103,29 +112,38 @@ def _wrap_with_scope_style(self, content: str, scope: str, emphasize: bool = Fal
 
 
 class TextChangeListener(sublime_plugin.TextChangeListener):
-    def __init__(self, buffer: Any, listener: AbstractViewListener) -> None:
-        super().__init__(buffer)
-        self.listener = ref(listener)
 
-    # Created to have symmetry with "attach". Weird naming in internal API...
-    def detach(self) -> None:
-        self.remove()
+    ids_to_listeners = WeakValueDictionary()  # type: WeakValueDictionary[int, TextChangeListener]
+
+    @classmethod
+    def is_applicable(cls, buffer: sublime.Buffer) -> bool:
+        views = buffer.views()
+        if views:
```

Comment:
Is the return value of `views()` not always a list? We can't just directly iterate?